.github/workflows/gh_pages.yml

```diff
@@ -1,30 +1,30 @@
 name: Github Pages
 on:
-  release:
-    types: [published]
+  push:
+    branches:
+      - master
 jobs:
   build-and-deploy:
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout
-        uses: actions/checkout@v2.3.1
-      
+      - uses: actions/checkout@v2.3.1
+
       - name: Install Node (14.x)
         uses: actions/setup-node@v2
         with:
           node-version: '14.x'
 
       - name: Install Dependencies
         run: yarn install
-        
+
       - name: Build
         run: gulp
 
       - name: Generate JSDoc
-        run: yarn jsdoc -c .jsdoc.json --verbose
+        run: yarn docs
 
       - name: Deploy to GH Pages
-        uses: JamesIves/github-pages-deploy-action@0f24da7de3e7e135102609a4c9633b025be8411b
```

Comment:
❗  Since this is not a GH official action we should be pointing to the commit hash rather than a tag version, as recommended by [github](https://docs.github.com/en/actions/learn-github-actions/security-hardening-for-github-actions#using-third-party-actions):

>**Pin actions to a full length commit SHA**
>
>Pinning an action to a full length commit SHA is currently the only way to use an action as an immutable release. Pinning to a particular SHA helps mitigate the risk of a bad actor adding a backdoor to the action's repository, as they would need to generate a SHA-1 collision for a valid Git object payload.
>
>**Pin actions to a tag only if you trust the creator**
>
>Although pinning to a commit SHA is the most secure option, specifying a tag is more convenient and is widely used. If you’d like to specify a tag, then be sure that you trust the action's creators. The ‘Verified creator’ badge on GitHub Marketplace is a useful signal, as it indicates that the action was written by a team whose identity has been verified by GitHub. Note that there is risk to this approach even if you trust the author, because a tag can be moved or deleted if a bad actor gains access to the repository storing the action.

Reply:
This is a security measure as it is possible for a tag to change the commit sha it's pointing to.

Reply:
Good call! I should've asked why we used a commit hash instead of changing it haha. I'll point it to the v4.1.5 SHA, instead!